libs/ledger-live-common/src/families/evm/editTransaction/getEditTransactionStatus.ts

```diff
@@ -0,0 +1,153 @@
+type ValidationIssues = Partial<Record<ValidatedTransactionFields, Error>>;
+
+/**
+ * Validate an edited transaction and returns related errors and warnings
+ * Makes sure the updated fees are at least 10% higher than the original fees
+ * 10% isn't defined in the protocol, it's just how most nodes & miners implement it
+ * If the new transaction fees are less than 10% higher than the original fees,
+ * the transaction will be rejected by the network with a
+ * "replacement transaction underpriced" error
+ */
+const validateEditTransaction = ({
+  editType,
+  transaction,
+  transactionToUpdate,
+}: {
+  transaction: Transaction;
+  transactionToUpdate: Transaction;
+  editType?: EditType;
+}): {
+  errors: TransactionStatusCommon["errors"];
+  warnings: TransactionStatusCommon["warnings"];
+} => {
+  const errors: ValidationIssues = {};
+  const warnings: ValidationIssues = {};
+
+  if (!editType) {
+    return {
+      errors: {},
+      warnings: {},
+    };
+  }
+
+  const { maxFeePerGas, maxPriorityFeePerGas, gasPrice } = transactionToUpdate;
+  const {
+    maxFeePerGas: newMaxFeePerGas,
+    maxPriorityFeePerGas: newMaxPriorityFeePerGas,
+    gasPrice: newGasPrice,
+  } = transaction;
+
+  if (transaction.type === 2) {
+    if (!maxFeePerGas || !maxPriorityFeePerGas) {
+      throw new Error("maxFeePerGas and maxPriorityFeePerGas are required");
+    }
+
+    if (!newMaxFeePerGas || !newMaxPriorityFeePerGas) {
+      // This should already be handled in getTransactionStatus
+      return {
+        errors: {},
+        warnings: {},
+      };
+    }
+
+    const { maxFeePerGas: minNewMaxFeePerGas, maxPriorityFeePerGas: minNewMaxPriorityFeePerGas } =
+      getMinEip1559Fees({ maxFeePerGas, maxPriorityFeePerGas });
+
+    if (
+      newMaxFeePerGas.isLessThan(minNewMaxFeePerGas) ||
+      newMaxPriorityFeePerGas.isLessThan(minNewMaxPriorityFeePerGas)
+    ) {
+      errors.replacementTransactionUnderpriced = new ReplacementTransactionUnderpriced();
+
+      return {
+        errors,
+        warnings,
+      };
+    }
+  } else {
+    if (!gasPrice) {
+      throw new Error("gasPrice is required");
+    }
+
+    if (!newGasPrice) {
+      // This should already be handled in getTransactionStatus
+      return {
+        errors: {},
+        warnings: {},
+      };
+    }
+
+    const { gasPrice: minNewGasPrice } = getMinLegacyFees({ gasPrice });
+
+    if (newGasPrice.isLessThan(minNewGasPrice)) {
+      errors.replacementTransactionUnderpriced = new ReplacementTransactionUnderpriced();
+
+      return {
+        errors,
+        warnings,
+      };
+    }
+  }
+
+  return {
+    errors: {},
+    warnings: {},
+  };
+};
+
+const ERROR_NOT_ENOUGH_NFT_OWNED = new NotEnoughNftOwned();
+const ERROR_NOT_OWNED_NFT = new NotOwnedNft();
+const ERROR_AMOUNT_REQUIRED = new AmountRequired();
+
+// NOTE: this logic should ideally be done in getTransactionStatus if possible
+// (not sure it is)
+export const getEditTransactionStatus = ({
```

Comment:
shouldn't this be in `coin-evm` ?

Reply:
Yes probably, could be moved in `libs/coin-evm/src/getTransactionStatus.ts`. Which would imply moving `libs/ledger-live-common/src/families/evm/editTransaction/types.ts` and `libs/ledger-live-common/src/families/evm/editTransaction/getMinFees.ts` under `coin-evm` as well since it depends on them as well and we can't have a dependency to `ledger-live-common` in `coin-evm`

Will take a look at that and propose a patch, see which solution is better

Reply:
Moved `getEditTransactionStatus` as well as `getMinFees` under `coin-evm`

cf. 32c6670a2af79384d7cb2673d1bf6df28c1eb286